### PR TITLE
Revive the filmstrip

### PR DESCRIPTION
**Pull request by rhystmills**
## What does this change?
@paperboyo pointed out that @itsibitzi worked on a filmstrip [PR](https://github.com/guardian/grid/pull/2949) back in 2020. This seemed like a useful bit of functionality, and potentially related to our current OKR 'Help Grid users to manage images and use the most suitable imagery available' given that it's related to image discoverability.

This PR basically rebases that PR's changes, makes them compatible with the Grid today, and aims to provide the minimum possible code changes required to retain its functionality.

## Open questions
- Should we pick this PR up and get it past the finishing line this quarter? 
- Are there any bugs or strange bits of behaviour to sort?
- Do we need to make further changes? (E.g. currently the filmstrip only shows the currently loaded images from the lazy-loaded infinite feed, rather than loading more when you get to the end/beginning)
- Can we stick it behind some sort of feature switch for beta testing?

### CODE REVIEW

**Comment by paperboyo** [comment]
Woah! Feel free to abandon, I’m already excited. My take was expressed here:  https://github.com/guardian/grid/pull/2949#issuecomment-665671882.

TL;DR: do the underlying changes first without any UI work. Hardest part anyways!
